### PR TITLE
enh(ci): MON-32903-add-sccache

### PR DESCRIPTION
**Pull request by kabreu**
Add sccache : 
https://github.com/mozilla/sccache/tree/main

And store the cache in S3 : 
https://github.com/mozilla/sccache/blob/main/docs/S3.md

The S3 bucket is created by terraform in the Infra repository : 
https://github.com/centreon/Infrastructure/tree/main/aws-accounts/software-factory/collect-sccache

At this moment the user used for accessing the bucket is the same as the "robot-report" used by other workflow using centreon-collect runners.

### CODE REVIEW

File: .github/workflows/centreon-collect.yml [hunk 1]
```diff
@@ -59,10 +59,10 @@ jobs:
 
     steps:
       - name: Checkout sources
-        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        uses: actions/checkout@v4
```
Comment (sc979) [inline .github/workflows/centreon-collect.yml:62]: kindly could you revert these changes ?

**Review by sc979 — DISMISSED** [review]
kindly restore gh actions hashes

File: .github/workflows/package-collect.yml [hunk 2]
```diff
@@ -58,6 +64,25 @@ jobs:
...
 
+      - name: Install sccache
+        run: |
+          if [ "${{ matrix.package_extension }}" = "deb" ]; then
+            apt install -y wget
+          fi
+          if [ "${{ matrix.package_extension }}" = "rpm" ]; then
+            dnf install -y wget
+          fi
+          if [ "${{ matrix.arch }}" = "amd64" ]; then
```
Comment (kduret) [inline .github/workflows/package-collect.yml:75]:
```suggestion
          elif [ "${{ matrix.arch }}" = "amd64" ]; then
```

File: .github/workflows/package-collect.yml [hunk 3]
```diff
@@ -58,6 +64,25 @@ jobs:
       - name: Checkout sources
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
+      - name: Install sccache
+        run: |
+          if [ "${{ matrix.package_extension }}" = "deb" ]; then
+            apt install -y wget
```
Comment (kduret) [inline .github/workflows/package-collect.yml:70]:
```suggestion
            apt-get update
            apt-get install -y wget
```

**Review by kduret — COMMENTED** [review]
to validate sccache is working as expected, you should test following cases :
- make a small change in this [file](https://github.com/centreon/centreon-collect/blob/develop/broker/neb/inc/com/centreon/broker/neb/acknowledgement.hh) and check cache is still used
- duplicate this pull request and check cache is still used

A retention rule should probably be apply on S3 bucket to infinite size incrementation

**Comment by kabreu** [comment]
@kduret 
**Test case 1 OK.** 
Can be verified on commit d683205 with [This Run](https://github.com/centreon/centreon-collect/actions/runs/7398247347/job/20127006316?pr=1023) EL9 took only 5 minutes and we got cache hits:
Compile requests                     555
Compile requests executed            555
Cache hits                           523
Cache hits (C/C++)                   523
Cache misses                          32
Cache misses (C/C++)                  32

**Test case 2 OK.**
Can be verified on this [duplicate PR](https://github.com/centreon/centreon-collect/pull/1036) with [This Run](https://github.com/centreon/centreon-collect/actions/runs/7398247347/job/20127015455?pr=1036)
Full cache hits for every OS/arch, all jobs have finished in 5 min or less.

File: .github/workflows/package-collect.yml [hunk 4]
```diff
@@ -140,7 +171,7 @@ jobs:
       # set condition to true if artifacts are needed
       - if: ${{ false }}
         name: Upload package artifacts
-        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        uses: actions/upload-artifact@v3
```
Comment (sc979) [inline .github/workflows/package-collect.yml:174]:
```suggestion
        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
```

File: .github/workflows/centreon-collect.yml [hunk 5]
```diff
@@ -90,7 +90,7 @@ jobs:
   deliver-sources:
     runs-on: [self-hosted, common]
     needs: [get-version, package]
-    if: ${{ contains(fromJson('["stable"]'), needs.get-version.outputs.stability) && github.event_name != 'workflow_dispatch' }}
+    if: ${{ contains(fromJson('["stable"]'), needs.get-version.outputs.stability) }}
```
Comment (sc979) [inline .github/workflows/centreon-collect.yml:93]: shouldn't it be restored ?

File: .github/workflows/actionlint.yml [hunk 6]
```diff
@@ -5,6 +5,7 @@ concurrency:
   cancel-in-progress: true
 
 on:
+  workflow_dispatch:
```
Comment (kduret) [inline .github/workflows/actionlint.yml:8]:
```suggestion
```

File: .github/workflows/package-collect.yml [hunk 7]
```diff
@@ -105,11 +130,17 @@ jobs:
...
+                  -DCMAKE_CXX_COMPILER_LAUNCHER=${SCCACHE_PATH} \
                   .
 
           sudo ninja
         shell: bash
 
+      - name: Cache statistics
+        run: |
+          ${SCCACHE_PATH} --show-stats
+
```
Comment (kduret) [inline .github/workflows/package-collect.yml:143]:
```suggestion

      - name: Stop sccache server
        run: ${SCCACHE_PATH} --stop-server
        shell: bash
```

File: broker/neb/inc/com/centreon/broker/neb/acknowledgement.hh [hunk 8]
```diff
@@ -44,7 +44,6 @@ class acknowledgement : public io::data {
   acknowledgement(acknowledgement const& other);
   ~acknowledgement();
   acknowledgement& operator=(acknowledgement const& other);
-
```
Comment (kduret) [inline broker/neb/inc/com/centreon/broker/neb/acknowledgement.hh:47]: restore changes

File: .github/workflows/package-collect.yml [hunk 9]
```diff
@@ -105,11 +130,17 @@ jobs:
...
+                  -DCMAKE_C_COMPILER_LAUNCHER=${SCCACHE_PATH} \
+                  -DCMAKE_CXX_COMPILER_LAUNCHER=${SCCACHE_PATH} \
                   .
 
           sudo ninja
         shell: bash
 
+      - name: Cache statistics
+        run: |
+          ${SCCACHE_PATH} --show-stats
```
Comment (kduret) [inline .github/workflows/package-collect.yml:142]:
```suggestion
        run: ${SCCACHE_PATH} --show-stats
        shell: bash
```

File: broker/neb/inc/com/centreon/broker/neb/acknowledgement.hh [hunk 10]
```diff
@@ -44,7 +44,7 @@ class acknowledgement : public io::data {
...
+
```
Comment (kduret) [inline broker/neb/inc/com/centreon/broker/neb/acknowledgement.hh:47]:
```suggestion

```

**Review by kduret — APPROVED** [review]

**Review by sc979 — DISMISSED** [review]

**Review by bouda1 — DISMISSED** [review]

**Review by kduret — APPROVED** [review]

File: broker/neb/inc/com/centreon/broker/neb/acknowledgement.hh [hunk 11]
```diff
@@ -44,7 +44,6 @@ class acknowledgement : public io::data {
   acknowledgement(acknowledgement const& other);
   ~acknowledgement();
   acknowledgement& operator=(acknowledgement const& other);
```
Comment (kduret) [inline broker/neb/inc/com/centreon/broker/neb/acknowledgement.hh:46]:
```suggestion
  acknowledgement& operator=(acknowledgement const& other);

```

**Review by kduret — APPROVED** [review]

**Review by sc979 — APPROVED** [review]
